cql3/functions/functions.cc

```diff
@@ -117,26 +113,27 @@ functions::init() noexcept {
     return ret;
 }
 
-void functions_changer::add_function(shared_ptr<function> func) {
+void functions::add_function(shared_ptr<function> func) {
     if (find(func->name(), func->arg_types())) {
         throw std::logic_error(format("duplicated function {}", func));
     }
     _declared.emplace(func->name(), func);
 }
 
-template <typename F>
-void functions_changer::with_udf_iter(const function_name& name, const std::vector<data_type>& arg_types, F&& f) {
-    auto i = find_iter(name, arg_types);
-    if (i == _declared.end() || i->second->is_native()) {
+void functions::with_udf_iter(const function_name& name, const std::vector<data_type>& arg_types, std::function<void(declared_t::iterator)> f) {
+    auto cit = find_iter(name, arg_types);
```

Comment:
I think the variable name shouldn't carry the type information, especially if it can be expressed by the type, it resembles hungarian notation, so just `const auto` here IMHO would be a better choice.

Reply:
it's named cit because few lines below there is it. So const alone wouldn't suffice. I could name it it1, it2, but not sure if this would be an improvement